src/App.js

```diff
@@ -1,61 +1,68 @@
-import { useState } from 'react';
 import useLocalStorage from './hooks/useLocalStorage.js';
 import Navigation from './Navigation.js';
 import WishlistPage from './pages/WishlistPage.js';
+import BookmarksPage from './pages/BookmarksPage.js';
 import AddWishPage from './pages/AddWishPage.js';
 
 export default function App() {
   const [diveWishes, setDiveWishes] = useLocalStorage('DivingWishlist', []);
-  const [isDialogVisible, setIsDialogVisible] = useState(false);
-  const [currentWishId, setCurrentWishId] = useState('');
 
   return (
     <AppGrid>
-      <Header>
-        <h1>Diving Wishlist</h1>
-      </Header>
-      <main>
-        <Routes>
-          <Route
-            path="/"
-            element={
-              <WishlistPage
-                diveWishes={diveWishes}
-                confirmDeleteWish={confirmDeleteWish}
-                cancelDeleteWish={cancelDeleteWish}
-                isDialogVisible={isDialogVisible}
-                showDeleteDialog={showDeleteDialog}
-              />
-            }
-          />
-          <Route
-            path="/add-wish"
-            element={<AddWishPage onAddDiveWish={addToWishlist} />}
-          />
-        </Routes>
-      </main>
+      <Routes>
+        <Route
+          path="/"
+          element={
+            <WishlistPage
+              diveWishes={diveWishes}
+              onToggleBookmark={toggleBookmark}
+              onDeleteDiveWish={handleDeleteWish}
+            />
+          }
+        />
+        <Route
+          path="/favorites"
+          element={
+            <BookmarksPage
+              bookmarkedWishes={diveWishes.filter(
+                diveWish => diveWish.isBookmarked === true
+              )}
+              onToggleBookmark={toggleBookmark}
+              onDeleteDiveWish={handleDeleteWish}
+            />
+          }
+        />
+        <Route
+          path="/add-wish"
+          element={<AddWishPage onAddDiveWish={handleAddWish} />}
+        />
+      </Routes>
       <Navigation />
     </AppGrid>
   );
 
-  function addToWishlist({ destination, notes }) {
+  function handleAddWish({ destination, notes }) {
     const id = nanoid();
     setDiveWishes([{ id, destination, notes }, ...diveWishes]);
   }
 
-  function confirmDeleteWish() {
-    setIsDialogVisible(false);
-    setDiveWishes(diveWishes.filter(diveWish => diveWish.id !== currentWishId));
+  function handleDeleteWish(id) {
+    setDiveWishes(diveWishes.filter(diveWish => diveWish.id !== id));
   }
-  function cancelDeleteWish() {
-    setIsDialogVisible(false);
-  }
-  function showDeleteDialog(id) {
-    setIsDialogVisible(true);
-    setCurrentWishId(id);
+
+  function toggleBookmark(id) {
+    setDiveWishes(
+      diveWishes.map(diveWish => {
+        if (id === diveWish.id) {
+          return { ...diveWish, isBookmarked: !diveWish.isBookmarked };
+        } else {
+          return diveWish;
+        }
+      })
+    );
   }
```

Comment:
You might consider using `immer` for state updates. I know Jerry prefers it. Again, this is personal preference :)